src/main/scala/is/hail/expr/ir/IR.scala

```diff
@@ -34,6 +34,13 @@ case class ApplyUnaryPrimOp(op: UnaryOp, x: IR, var typ: Type = null) extends IR
 
 case class MakeArray(args: Array[IR], var typ: TArray = null) extends IR {
   override def toString(): String = s"MakeArray(${args: IndexedSeq[IR]}, $typ)"
+  override def equals(that: Any): Boolean = that match {
```

Comment:
You can't override `equals` without `hashCode`.  Or make `args` an `IndexedSeq`.

Reply:
Ah oops, I had the implication between hashcode and equals backwards

Reply:
Scala makes implementation of hashCode rather easy, let me know what you think of the above.